mrmustard/lab_dev/circuit_components.py

```diff
@@ -552,7 +646,7 @@ def __rrshift__(self, other: Scalar) -> CircuitComponent | np.array:
             return ret
 
     def __repr__(self) -> str:
-        return f"{self.__class__.__name__}(modes={self.modes}, name={self.name or None})"
+        return f"{self.__class__.__name__}(modes={self.modes}, name={self.name or None}, repr={self.representation.__class__.__name__})"
```

Comment:
can we only add this if the representation is defined? the tests show `repr=NoneType` which is kinda noisy for users

Reply:
yes, good idea